text/0517-io-os-reform.md

```diff
@@ -483,7 +483,72 @@ throughout IO, we can go on to explore the modules in detail.
 ### `std::env`
 [std::env]: #stdenv
 
-> To be added in a follow-up PR.
+Most of what's available in `std::os` today will move to `std::env`,
+and the signatures will be updated to follow this RFC's
+[Design principles] as follows.
+
+**Arguments**:
+
+* `args`: change to yield an iterator rather than vector if possible; in any case, it should produce an `OsStrBuf`.
+
+**Environment variables**:
+
+* `vars` (renamed from `env`): yields a vector of `(OsStrBuf, OsStrBuf)` pairs.
+* `var` (renamed from `getenv`): take a value bounded by `IntoOsStrBuf`,
```

Comment:
I think it would make more sense for these APIs to return a `Result<String, EnvError>` where `EnvError` is:

```rs
enum EnvError {
  Missing,
  NotUnicode(OsStrBuf)
}
```

That way you can quickly and ergonomically get at a String, which is usually what you want, but still get at the `OsStrBuf` if you really need it (usually to present a good error message).